### PR TITLE
70 save user password/ register a new user!!

### LINKED ISSUE #70: [3pts] App Backend: Save User & Password Login
**Fill in the blanks**:
As a developer, I would like to have a helper method that saves the user login, so that the method can be reused.

**Describe the solution you'd like:**
- [ ] Parameter(s): login string
- [ ] Return: void

**Definition of Done - Check as you go!**
- [ ] Tasks have been implemented
- [ ] Documented
- [ ] Passing Tests for each task
- [ ] Pull Request made
- [ ] Code Reviewed

### PR DESCRIPTION
**Pull request by brendacano**
register user, connected to db

### CODE REVIEW

File: app_backend/server.py [hunk 1]
```diff
@@ -56,5 +59,76 @@ async def add_friend(resp: Response, uuid: str = None, fuuid: str = None):
...
+            The status code for the request        
+"""
+@app.post("/register/", status_code=status.HTTP_201_CREATED)
+async def registeruser(login: UserLogin, resp: Response):
+
+    username = login.username
+    email = login.email
+    password = login.password
+
+    validUsername = r'^(?=[a-zA-Z0-9._]{8,20}$)(?!.*[_.]{2})[^_.].*[^_.]$'
```
Comment (MimiOlayeye) [inline app_backend/server.py:109]: Can you add a comment explaining what these values are, I'm not familiar  @brendacano

File: app_backend/user.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+user_db = db.client["user"]
+userinfo_collection = user_db["userlogininfo"]
+resp = Response
+
+class User():
+    def save_user_in_db(loginInfo: dict):
+        for eml in userinfo_collection.find({}, {"_id": 0, "username": 0, "password": 0}):
+            if loginInfo['email'] == eml['email']:
+                resp.status_code = status.HTTP_400_BAD_REQUEST
+                return {"err" : "Email already exists. Use a different email."}
```
Comment (RafaelPiloto10) [inline app_backend/user.py]: This `save_user_in_db` function looks like it pulls all users and manually checks all the emails to check for duplicates. Instead, is it possible to query for all entries with that email and then check if the size of our result is greater than 0? If so, that means we found a user with the same email.

This decrease the amount of data communicated by the database and also takes advantage of the quick lookup tables and fancy algorithms MongoDB probably has.

File: app_backend/server_test.py [hunk 3]
```diff
@@ -32,3 +32,35 @@ def __init__(self, uuid: str, fuuid: str, expected_status_code: int, expects_err
...
+            self.password = password
+            self.expects_err = expects_err
+            self.expects_err_msg = expects_err_msg
+        def to_json(self):
+            return {"username": self.username, "email": self.email, "password": self.password}
+    test_cases = [
+        Params('test000', 'test0@emory.edu', 'Test0000', status.HTTP_400_BAD_REQUEST, {"err" : "Invalid username. It must be 8 to 20 characters with no special characters."}),
+        Params('test0000', 'test0@gmail.com', 'Test0000', status.HTTP_400_BAD_REQUEST, {"err" : "Invalid email address. It must be an emory.edu email."}),
+        Params('test0000', 'test0@emory.edu', 'test0000', status.HTTP_400_BAD_REQUEST, {"err" : "Invalid password. It must be 8 to 20 characters and have at least one upper case letter, one lower case letter, and one digit."}),
+        Params('test0000', 'test@emory.edu', 'Test0000', status.HTTP_201_CREATED, {"msg": "Successfully registered new user."})
```
Comment (RafaelPiloto10) [inline app_backend/server_test.py]: I don't really care for it much at the moment, but we should add more diverse combinations here to verify the regex you built works

**Review by RafaelPiloto10 — CHANGES_REQUESTED** [review]
Groundbreaking work on this! On small thing but it LGTM

File: app_backend/server.py (hunk 1 above)
Comment (brendacano) [reply app_backend/server.py:109]: Will do

File: app_backend/user.py (hunk 2 above)
Comment (brendacano) [reply app_backend/user.py]: Good idea on both the comments you made. There is definetly to do this

Comment (brendacano) [reply app_backend/user.py]: changed it please check when you can

File: app_backend/server_test.py (hunk 3 above)
Comment (brendacano) [reply app_backend/server_test.py]: i added a few more tests, hopefully could do more as I think of more test cases while finishing up the /login/ route

File: app_backend/server.py (hunk 1 above)
Comment (brendacano) [reply app_backend/server.py:109]: done, please check if this is what you had in mind

**Review by RafaelPiloto10 — APPROVED** [review]
Nice work! LGTM

**Review by Foxworth22 — APPROVED** [review]
Looks wonderful! (I'm guessing from the !! in the title you are excited about it too :)

**Pull request merged**